lib/client/ipam_block_reader_writer.go

```diff
@@ -297,6 +297,13 @@ func blockGenerator(pool cnet.IPNet) func() *cnet.IPNet {
 	}
 }
 
+// initialIndex keeps track of the random starting point
```

Comment:
I don't think you want these defined _outside_ of randomBlockGenerator.  They should be initialized each time it is called.